static/tapered/takum/arithmetic/wide_arithmetic.cpp

```
@@ -0,0 +1,513 @@
+
+// ---------------------------------------------------------------------------
+// a OP b, correctly rounded
+// ---------------------------------------------------------------------------
+template<unsigned nbits, unsigned rbits>
+int VerifyCorrectlyRounded(Op op, std::uint64_t stride, bool reportTestCases) {
+	using T = sw::universal::takum<nbits, rbits, std::uint64_t>;
+	int nrOfFailedTests = 0;
+	long exercised = 0;
+	// 1ull << nbits is undefined at nbits == 64, and 64 is the width this exists
+	// for, so the bound avoids it.  stride samples, since the sweep is over PAIRS.
+	const std::uint64_t NR = (nbits >= 64) ? ~0ull : (1ull << nbits);
```

Cppcheck (2.21.0)

[error] Shifting 64-bit value by 64 bits is undefined behaviour

(shiftTooManyBits)